appendable-rs/appendable/src/tests/jsonl_index_file.rs

```diff
@@ -0,0 +1,39 @@
+#[cfg(test)]
+mod tests {
+    use crate::handler::jsonl_handler::JSONLHandler;
+    use crate::index_file::IndexFile;
+    use std::fs::File;
+    use std::io::Write;
+    use std::path::Path;
+    use tempfile::NamedTempFile;
+
+    fn mock_jsonl_file() -> std::io::Result<File> {
+        // Create a temporary file
+        let mut temp_file = NamedTempFile::new()?;
+
+        writeln!(
+            temp_file,
+            r#"{{"name": "matteo", "id": 2, "alpha": ["a", "b", "c"]}}"#
+        )?;
+        writeln!(
+            temp_file,
+            r#"{{"name": "kevin", "id": 1, "alpha": ["x", "y", "z"]}}"#
+        )?;
+
+        // Persist the file and return the File handle
+        let file = temp_file.persist(Path::new("mock_data.jsonl"))?;
```

Comment:
Is it possible to do this in memory without having to write a file?

Reply:
Yes this makes sense. However there was a reason why I wanted to persist the file right now.

That is to feed in the `mock_data.jsonl` to our existing Go engine. The goal is to add detailed print statements upon index file creation and do a deep comparison between our Rust and Go versions.

Would it be alright to keep this, create another version that's in memory, then when the Rust version is mature we can remove the above code?

Reply:
Yeah that's fine